packages/mui-system/src/Unstable_Grid/gridGenerator.ts

```diff
@@ -31,6 +31,13 @@ function createGetParentSpacing(ownerState: Props['ownerState']) {
   };
 }
 
+function getParentColumns(ownerState: Props['ownerState']) {
+  if (ownerState.level === 0) {
+    return `var(--Grid-columns)`;
+  }
+  return `var(--Grid-columns${ownerState.level - 1 || ''})`;
```

Comment:
nit, I would name the variable to be `--Grid-columns-level1` instead of `--Grid-columns1`.

Reply:
As per https://github.com/mui/material-ui/issues/35474#issue-1496175071, it should be `--Grid-columnsLevel1`.